detoks/src/main/java/nl/tudelft/trustchain/detoks/DeToksCommunity.kt

```diff
@@ -99,11 +123,79 @@ class DeToksCommunity(private val context: Context) : Community() {
         }
     }
 
+    override fun walkTo(address: IPv4Address) {
+        super.walkTo(address)
+
+        discoveredAddressesContacted[address] = Date()
+    }
+
+    override fun onIntroductionResponse(peer: Peer, payload: IntroductionResponsePayload) {
+        super.onIntroductionResponse(peer, payload)
+
+        if (peer.address in DEFAULT_ADDRESSES) {
+            lastTrackerResponses[peer.address] = Date()
+        }
+    }
+
+    fun broadcastLike(vid: String, torrent: String, creator: String) {
+        Log.d("DeToks", "Liking: $vid")
+
+        // TODO: fix this
+//        val peer = (0 until getPeers().size).random()
+//        val peerKey = getPeers()[peer].key.keyToBin()
+        val like = Like(myPeer.publicKey.toString(), vid, torrent, creator)
+        createProposalBlock(LIKE_BLOCK, like.toMap(), myPeer.publicKey.keyToBin())
+        // createProposalBlock(LIKE_BLOCK, like.toMap(), peerKey)
+        Log.d("DeToks", "$like")
+    }
+
+    fun getLikes(vid: String, torrent: String): List<TrustChainBlock> {
+        return database.getBlocksWithType(LIKE_BLOCK).filter {
+            it.transaction["video"] == vid && it.transaction["torrent"] == torrent
+        }
+    }
+
+    fun getBlocksByAuthor(author: String): List<TrustChainBlock> {
+        return database.getBlocksWithType(LIKE_BLOCK).filter {
+            it.transaction["author"] == author
+        }
+    }
+
+    fun userLikedVideo(vid: String, torrent: String, liker: String): Boolean {
+        return getLikes(vid, torrent).filter { it.transaction["liker"] == liker }.isNotEmpty()
```

Comment:
You can use 
 
return liker in getLikes(...)

Or 

return getLikes(...).contains(liker)

This doesn't seem like the most idiomatic approach, and I'm pretty sure it should be slower and also harder to read